Assets/DanmakU/Runtime/Core/Danmaku.cs

```diff
@@ -51,6 +51,13 @@ public struct Danmaku {
     [MethodImpl(MethodImplOptions.AggressiveInlining)] get { return Pool.Positions[Id]; }
     [MethodImpl(MethodImplOptions.AggressiveInlining)] set { Pool.Positions[Id] = value; }
   }
+  /// <summary>
+  /// Gets or sets the world position of the Danmaku during the previous frame.
+  /// </summary>
+  public Vector2 OldPosition {
+    [MethodImpl(MethodImplOptions.AggressiveInlining)] get { return Pool.OldPositions[Id]; }
+    [MethodImpl(MethodImplOptions.AggressiveInlining)] private set { Pool.OldPositions[Id] = value; }
+  }
```

Comment:
This should not be a public value to change at a per-Danmaku level: OldPosition should be an internal only value. Please change the access modifier.

Reply:
The setter is already private, but I'll remove the getter.

Reply:
(Unless you meant that the pool shouldn't expose OldPositions....)

Reply:
I think the getter is fine, but the overall access modifier of "public" is too broad for what is an internal property at the DanmakuPool level. Also, IMO, setter only properties generally are bad practice. Just change the access modifier to internal, and put back the getter.